src/main/java/org/mastodon/tomancak/PlainTextFileExport.java

```diff
@@ -0,0 +1,483 @@
+import org.scijava.ui.behaviour.util.Actions;
+import org.scijava.ui.behaviour.util.AbstractNamedAction;
+import org.scijava.ui.behaviour.util.RunnableAction;
+import org.scijava.log.LogLevel;
+import org.scijava.log.LogService;
+
+
+@Plugin( type = PlainTextFileExport.class )
+public class PlainTextFileExport extends AbstractContextual implements MastodonPlugin
+{
+	//"IDs" of all plug-ins wrapped in this class
+	private static final String TXT_EXPORT = "[tomancak] export tracks as txt file";
+	//------------------------------------------------------------------------
+
+	@Override
+	public List< ViewMenuBuilder.MenuItem > getMenuItems()
+	{
+		//this places the plug-in's menu items into the menu,
+		//the titles of the items are defined right below
+		return Arrays.asList(
+				menu( "Plugins",
+						menu( "Tomancak lab",
+							item( TXT_EXPORT ) ) ) );
+	}
+
+	/** titles of this plug-in's menu items */
+	private static Map< String, String > menuTexts = new HashMap<>();
+	static
+	{
+		menuTexts.put( TXT_EXPORT, "Export to plain text file" );
+	}
+
+	@Override
+	public Map< String, String > getMenuTexts()
+	{
+		return menuTexts;
+	}
+	//------------------------------------------------------------------------
+
+	private final AbstractNamedAction actionExport;
+
+	/** default c'tor: creates Actions available from this plug-in */
+	public PlainTextFileExport()
+	{
+		actionExport = new RunnableAction( TXT_EXPORT, this::exporter );
+		updateEnabledActions();
+	}
+
+	/** register the actions to the application (with no shortcut keys) */
+	@Override
+	public void installGlobalActions( final Actions actions )
+	{
+		final String[] noShortCut = new String[] { "not mapped" };
+		actions.namedAction( actionExport, noShortCut );
+	}
+
+	/** reference to the currently available project in Mastodon */
+	private MastodonPluginAppModel pluginAppModel;
+
+	/** learn about the current project's params */
+	@Override
+	public void setAppModel( final MastodonPluginAppModel model )
+	{
+		//the application reports back to us if some project is available
+		this.pluginAppModel = model;
+		updateEnabledActions();
+	}
+
+	/** enables/disables menu items based on the availability of some project */
+	private void updateEnabledActions()
+	{
+		final MamutAppModel appModel = ( pluginAppModel == null ) ? null : pluginAppModel.getAppModel();
+		actionExport.setEnabled( appModel != null );
+	}
+	//------------------------------------------------------------------------
+	private LogService logServiceRef; //(re)defined with every call to this.exporter()
+
+
+	/** opens the output file dialog, runs the export,
+	    and pops-up a "done+hints" message window */
+	private void exporter()
+	{
+		//open a folder choosing dialog
+		final File selectedFile = FileChooser.chooseFile(null, null,
+				new ExtensionFileFilter("txt"),
+				"Choose TXT file to write tracks into:",
+				FileChooser.DialogType.SAVE,
+				FileChooser.SelectionMode.FILES_ONLY);
```

Comment:
You could use [`UIService#chooseFile()`](https://javadoc.scijava.org/SciJava/index.html?org/scijava/ui/UIService.html) here (and improve the SciJava API in case it doesn't meet your requirements ;-)

Reply:
thanks @imagejan... I actually did not know about it and was anyway blindly following example of how to open file from some other Mastodon code,
I'll fix it... on my TODO list now ;-)